inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/service/ConfigService.java

```diff
@@ -0,0 +1,228 @@
+                TimeUnit.SECONDS);
+    }
+
+    /**
+     * Create data source.
+     */
+    private void createDataSource() {
+        JdbcConfig jdbcConfig = JdbcUtils.buildMysqlConfig();
+        HikariConfig config = new HikariConfig();
+        config.setDriverClassName(jdbcConfig.getDriverClass());
+        config.setJdbcUrl(jdbcConfig.getJdbcUrl());
+        config.setUsername(jdbcConfig.getUserName());
+        config.setPassword(jdbcConfig.getPassword());
+        config.setConnectionTimeout(Configuration.getInstance().get(KEY_DATASOURCE_CONNECTION_TIMEOUT,
+                DEFAULT_CONNECTION_TIMEOUT));
+        config.addDataSourceProperty(CACHE_PREP_STMTS,
+                Configuration.getInstance().get(KEY_CACHE_PREP_STMTS, DEFAULT_CACHE_PREP_STMTS));
+        config.addDataSourceProperty(PREP_STMT_CACHE_SIZE,
+                Configuration.getInstance().get(KEY_PREP_STMT_CACHE_SIZE, DEFAULT_PREP_STMT_CACHE_SIZE));
+        config.addDataSourceProperty(PREP_STMT_CACHE_SQL_LIMIT,
+                Configuration.getInstance().get(KEY_PREP_STMT_CACHE_SQL_LIMIT, DEFAULT_PREP_STMT_CACHE_SQL_LIMIT));
+        config.setMaximumPoolSize(
+                Configuration.getInstance().get(KEY_DATASOURCE_POOL_SIZE,
+                        DEFAULT_DATASOURCE_POOL_SIZE));
+        dataSource = new HikariDataSource(config);
+    }
+
+    /**
+     * Update audit ids.
+     */
+    private void updateAuditIds() {
+        try (Connection connection = dataSource.getConnection();
+                PreparedStatement pstat = connection.prepareStatement(queryAuditIdSql)) {
+            if (connection.isClosed()) {
+                createDataSource();
+            }
+
+            try (ResultSet resultSet = pstat.executeQuery()) {
+                CopyOnWriteArrayList<String> auditIdList = new CopyOnWriteArrayList<>();
+                while (resultSet.next()) {
+                    String auditId = resultSet.getString(1);
+                    LOGGER.info("Update audit id {}", auditId);
+                    auditIdList.add(auditId);
+                }
+                if (!auditIdList.isEmpty()) {
+                    auditIds = auditIdList;
+                }
+            } catch (SQLException sqlException) {
+                LOGGER.error("Query has SQL exception! ", sqlException);
+            }
+
+        } catch (Exception exception) {
+            LOGGER.error("Query has exception! ", exception);
+        }
+    }
+
+    /**
+     * Update audit source.
+     */
+    private void updateAuditSource() {
+        try (Connection connection = dataSource.getConnection();
+                PreparedStatement pstat = connection.prepareStatement(queryAuditSourceSql)) {
+            if (connection.isClosed()) {
+                createDataSource();
+            }
+            try (ResultSet resultSet = pstat.executeQuery()) {
+                ConcurrentHashMap<String, List<JdbcConfig>> auditSource = new ConcurrentHashMap<>();
```

Comment:
A HashMap is enough

Reply:
The member variable auditSources may have multi-thread concurrency. In order to facilitate the replacement of member variables, the temporary variable here adopts the same type as the member variable of ConcurrentHashMap.